lib/afc/index.js

```diff
@@ -20,7 +20,7 @@ class AfcService extends BaseServiceSocket {
   constructor (socketClient) {
     super(socketClient);
 
-    this._splitter = new LengthBasedSplitter(true, MB, 8, 8, -8);
+    this._splitter = new LengthBasedSplitter(socketClient, true, MB, 8, 8, -8);
```

Comment:
The constructor of `LengthBasedSplitter` accepts too many params. Consider packing them into an object

Reply:
I agree. Though it’s probably better done in a refactoring PR than within a feature one.